monai/bundle/scripts.py

```diff
@@ -102,11 +108,71 @@ def run(
     parser.read_meta(f=_args.pop("meta_file"))
     id = _args.pop("target_id", "")
 
-    # the rest key-values in the args are to override config content
+    # the rest key-values in the _args are to override config content
     for k, v in _args.items():
         parser[k] = v
 
     workflow = parser.get_parsed_content(id=id)
     if not hasattr(workflow, "run"):
         raise ValueError(f"The parsed workflow {type(workflow)} does not have a `run` method.\n{run.__doc__}")
     workflow.run()
+
+
+def verify_metadata(
+    meta_file: Optional[Union[str, Sequence[str]]] = None,
+    schema_url: Optional[str] = None,
+    filepath: Optional[PathLike] = None,
+    result_path: Optional[PathLike] = None,
+    create_dir: Optional[bool] = None,
+    hash_val: Optional[str] = None,
+    args_file: Optional[str] = None,
+    **kwargs,
+):
+    """
+    Verify the provided `metadata` file based on the predefined `schema`.
+    The schema standard follows: http://json-schema.org/.
+
+    Args:
+        meta_file: filepath of the metadata file to verify, if `None`, must be provided in `args_file`.
+            if it is a list of file paths, the content of them will be merged.
+        schema_url: URL to download the expected schema file.
+        filepath: file path to store the downloaded schema.
+        result_path: if not None, save the validation error into the result file.
+        create_dir: whether to create directories if not existing, default to `True`.
+        hash_val: if not None, define the hash value to verify the downloaded schema file.
+        args_file: a JSON or YAML file to provide default values for all the args in this function.
+            so that the command line inputs can be simplified.
+        kwargs: other arguments for `jsonschema.validate()`. for more details:
+            https://python-jsonschema.readthedocs.io/en/stable/validate/#jsonschema.validate.
+
+    """
+
+    k_v = zip(
+        ["meta_file", "schema_url", "filepath", "result_path", "create_dir", "hash_val"],
+        [meta_file, schema_url, filepath, result_path, create_dir, hash_val],
+    )
+    for k, v in k_v:
```

Comment:
There are duplicated code in the scripts entry point, could you help reduce the number of lines by having new utilities?

Reply:
Sure, optimized in the latest commit.

Thanks.